sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -253,7 +250,7 @@ case class Cast(child: Expression, dataType: DataType) extends UnaryExpression w
 
   // Converts Date to string according to Hive DateWritable convention
   private[this] def dateToString(d: Date): String = {
```

Comment:
if Date already has a good toString implementaiton, maybe you can remove this function entirely?